### ISSUE TITLE
A newly detected buggy behavior with c++ compilers

### ISSUE THREAD

**Issue by stefanos82** [edubart/nelua-lang#281]
@edubart I have compiled the latest changes after you reported the bug fix in #279 (thank you for that!), I have decided to check all my files, one my one.

Remember this gist of yours, where you demonstrate Celeste version implemented in Nelua?

https://gist.github.com/edubart/a79bf78a249d1fff2b77728c260c7605

Well, this one throws some interesting error messages; seems like behind the scenes, you are using the word `this` as a parameter name and both GCC (`g++`) and Clang (`clang++`) stop the compilation due to the usage of `this` keyword.

**Comment by edubart** [comment]
Makes sense, there are a lot of C++ reserved keywords that need to be listed to avoid problems like this at https://github.com/edubart/nelua-lang/blob/master/lualib/nelua/cdefs.lua#L649 . I've listed all keywords relevant C11, but indeed words related to C++ must be added there, will try to do this later.
